### PR TITLE
Refined destroy

### PR DESCRIPTION
**Pull request by chezzdev**

### CODE REVIEW

File: MapboxVisionAR/VisionARManager.swift [hunk 1]
```diff
@@ -11,42 +11,35 @@ import MapboxVisionCore
...
-        native.setRoute(route)
+    public static func create(visionManager: VisionManager, delegate: VisionARDelegate? = nil) -> VisionARManager {
+        let manager = VisionARManager()
+        manager.native = VisionARManagerNative.create(visionManager: visionManager.native, delegate: manager)
+        manager.delegate = delegate
+        return manager
     }
     
     public func destroy() {
-        guard let native = native else {
```
Comment (vanmaxim) [inline MapboxVisionAR/VisionARManager.swift]: Why you remove this check?

Comment (chezzdev) [reply MapboxVisionAR/VisionARManager.swift]: Didn't really get its purpose. Was it needed to guard from incorrect configuration in `create`?

Comment (vanmaxim) [reply MapboxVisionAR/VisionARManager.swift]: I think we need to know (by assert) what something is going wrong. For example: We trying to call 'setRoute' after 'destroy'.

Comment (chezzdev) [reply MapboxVisionAR/VisionARManager.swift]: Now that `native` is being nulled in `destroy`, we may check for multiple `destroy` calls to warn user about incorrect lifecycle management.
On the other hand, setter methods on "destroyed" manager producing no-op, seems fine to me.

**Pull request merged**